src/Workspaces/CoreTest/Renamer/RenamerTests.VisualBasic.cs

```diff
@@ -0,0 +1,213 @@
+            expectRename: false);
+
+        [Fact]
+        public Task VisualBasic_RenameDocument_RenameType()
+        => TestRenameDocumentVisualBasic(
+$@"Class {DefaultDocumentName}
+End Class",
+@"Class NewDocumentName
+End Class",
+            newDocumentName: "NewDocumentName");
+
+        [Fact]
+        public Task VisualBasic_RenameDocument_RenameInterface()
+        => TestRenameDocumentVisualBasic(
+@"Interface IInterface
+End Interface",
+@"Interface IInterface2
+End Interface",
+            documentName: "IInterface.vb",
+            newDocumentName: "IInterface2.vb");
+
+
+        [Fact]
+        public Task VisualBasic_RenameDocument_RenameEnum()
+        => TestRenameDocumentVisualBasic(
+            @"enum MyEnum {}",
+            @"enum MyEnum2 {}",
+            documentName: "MyEnum.vb",
+            newDocumentName: "MyEnum2.vb");
+
+        [Fact]
+        public Task VisualBasic_RenameDocument_RenamePartialClass()
+        {
+            var originalDocuments = new[]
+            {
+                new DocumentWithInfo()
+                {
+                    Text = @"
+Namespace Test
+    Partial Class C
+    End Class
+End Namespace",
+                    DocumentFilePath = @"Test\Folder\Path\C.vb",
+                    DocumentName = "C.vb"
+                },
+                new DocumentWithInfo()
+                {
+                    Text = @"
+Namespace Test
+    Partial Class C
+        Class Other
+        End Class
+    End Class
+End Namespace",
+                    DocumentFilePath = @"Test\Folder\Path\C.Other.vb",
+                    DocumentName = "C.Other.vb"
+                }
+            };
+
+            var expectedDocuments = new[]
+            {
+                new DocumentWithInfo()
+                {
+                    Text = @"
+Namespace Test
+    Partial Class C2
+    End Class
+End Namespace",
+                    DocumentFilePath = @"Test\Folder\Path\C2.vb",
+                    DocumentName = "C2.vb",
+                    ExpectRename = true
+                },
+                new DocumentWithInfo()
+                {
+                    Text = @"
+Namespace Test
+    Partial Class C2
+        Class Other
+        End Class
+    End Class
+End Namespace",
+                    DocumentFilePath = @"Test\Folder\Path\C.Other.vb",
+                    DocumentName = "C.Other.vb"
+                }
+            };
+
+            return TestRenameDocumentVisualBasic(originalDocuments, expectedDocuments);
+        }
+
+
+        [Fact]
+        public Task VisualBasic_RenameDocument_NoRenameNamespace()
+        => TestRenameDocumentVisualBasic(
+@"Namespace Test.Path
+    Class C
+    End Class
+End Namespace",
+@"Namespace Test.Path
+    Class C
+    End Class
+End Namespace",
+        documentPath: @"Test\Path\Document.vb",
+        documentName: @"Document.vb",
+        expectRename: false);
+
+        // Enabling these tests is tracked by https://github.com/dotnet/roslyn/issues/41841
```

Comment:
Leave the test uncommented and skip it with a skip attribute instead? Easier to find (and automate the finding) of tests that need to be fixed once the bug is closed.